### PR TITLE
Movement: Increased GroundFriction, decreased sliding, Added Stamina

### PR DESCRIPTION
**Pull request by Alf21**
Just a suggestion, imo it feels weird as it is such slower now, but maybe better for TTT. Test around if you want :)

image

### CODE REVIEW

**Review by mzegar — APPROVED** [review]
The change is fine, but I wonder if we should just leave changes like this out until we fully decide to create our on `WalkController` and then change more values.

File: code/player/TTTPlayer.cs [hunk 1]
```diff
@@ -10,6 +10,12 @@ public partial class TTTPlayer : Sandbox.Player
     {
         private static int CarriableDropVelocity { get; set; } = 300;
 
+        [Net, Predicted]
+        public float Stamina { get; set; } = 100f;
+
+        [Net]
+        public float MaxStamina { get; set; } = 100f;
```
[the remark this answers is not in the document]
Comment (Alf21) [reply code/player/TTTPlayer.cs:17]: Agreed, but without this, sprinting is still possible. That's why I added Stamina to it. Anyways, exporting already existing things is better than waiting for a sbox change and then doing it (if it's already needed to balance the game).

[the remark this answers is not in the document]
Comment (mzegar) [reply code/player/TTTPlayer.cs:17]: Right, we could just disable sprinting then.

And I would probably agree that waiting isn't the best idea... but my two thoughts are...
1. We are going to need to tear into multiple files to extract features such as "sprinting" whenever addon support is done, potentially breaking other things.
2. It's not like we've ran out of things to do, we have plenty of other core features that need to be handled. I would agree waiting wouldn't make sense if we had nothing else to implement.

Anyways, just my two cents, personally I'd just disable sprinting and leave any "extra" features until we have nothing else to do. The PR is approved, so up to you if you want to hear out other opinions or just merge this.

[the remark this answers is not in the document]
Comment (Alf21) [reply code/player/TTTPlayer.cs:17]: Okay I gonna add a ConVar to enable sprint and disable it by default. Anyways, this custom DefaultWalkController is needed for upcoming core things like Drowning and (improved) FallDamage detection.

@saibotk @AMastryukov @TimGoll

File: code/player/controller/DefaultWalkController.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public partial class DefaultWalkController : WalkController
+    {
+        [ServerVar]
+        public static bool IsSprintEnabled { get; set; } = false;
+
+        public float MaxSprintSpeed = 300f;
+        public float StaminaLossPerSecond = 30f;
+        public float StaminaGainPerSecond = 25f;
+
+        private bool _jumped = false;
```
Comment (mzegar) [inline code/player/controller/DefaultWalkController.cs:16]: Is this used anywhere?

**Review by mzegar — COMMENTED** [review]
Quick comment, seems reasonable to me.

File: code/player/controller/DefaultWalkController.cs (hunk 2 above)
Comment (Alf21) [reply code/player/controller/DefaultWalkController.cs]: ty for the hint, used in previous testing and I forget to remove it ^^

**Pull request merged**